modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManagerImpl.java

```diff
@@ -17,53 +17,282 @@
+                                    partitions
+                                ),
+                                new TableSchemaView() {
+                                    @Override public SchemaDescriptor schema() {
+                                        return schemaManager.schema(tblId);
+                                    }
+
+                                    @Override public SchemaDescriptor schema(int ver) {
+                                        return schemaManager.schema(tblId, ver);
+                                    }
+                                }));
+                        }
+                        catch (InterruptedException | ExecutionException e) {
+                            LOG.error("Failed to start table [key={}]",
+                                evt.newEntry().key(), e);
+                        }
+                    }
+                }
+
+                return false;
+            }
+
+            @Override public void onError(@NotNull Throwable e) {
+                LOG.error("Metastorage listener issue", e);
+            }
+        });
+    }
+
+    /**
+     * Checks whether the local node hosts Metastorage.
+     *
+     * @param localNodeName Local node uniq name.
+     * @param metastorageMembers Metastorage members names.
+     * @return True if the node has Metastorage, false otherwise.
+     */
+    private boolean hasMetastorageLocally(String localNodeName, String[] metastorageMembers) {
+        boolean isLocalNodeHasMetasorage = false;
+
+        for (String name : metastorageMembers) {
+            if (name.equals(localNodeName)) {
+                isLocalNodeHasMetasorage = true;
+
+                break;
+            }
+        }
+        return isLocalNodeHasMetasorage;
+    }
+
+    /**
+     * Subscribes on table create.
+     */
+    private void subscribeForTableCreation() {
+        configurationMgr.configurationRegistry().getConfiguration(TablesConfiguration.KEY)
+            .tables().listen(ctx -> {
+            HashSet<String> tblNamesToStart = new HashSet<>(ctx.newValue().namedListKeys());
+
+            long revision = ctx.storageRevision();
+
+            if (ctx.oldValue() != null)
+                tblNamesToStart.removeAll(ctx.oldValue().namedListKeys());
+
+            for (String tblName : tblNamesToStart) {
+                TableView tableView = ctx.newValue().get(tblName);
+                long update = 0;
+
+                UUID tblId = new UUID(revision, update);
+
+                CompletableFuture<Boolean> fut = metaStorageMgr.invoke(
+                    new Key(INTERNAL_PREFIX + tblId.toString()),
+                    Conditions.value().eq(null),
+                    Operations.put(tableView.name().getBytes(StandardCharsets.UTF_8)),
+                    Operations.noop());
+
+                try {
+                    if (fut.get()) {
+                        metaStorageMgr.put(new Key(INTERNAL_PREFIX + "assignment." + tblId.toString()), new byte[0]);
```

Comment:
It's important to update both NTERNAL_PREFIX + tblId, INTERNAL_PREFIX + "assignment" and others within one transactional update. @agura will provide 
`invoke(@NotNull Condition condition, @NotNull Collection<Operation> success, @NotNull Collection<Operation> failure);` for such purposes. So it'll be also available with metaStorageMgr.

Reply:
Added TODO with ticket. It will be done after the multi-invoke would be implemented.